earth_enterprise/src/common/khTypes.h

```diff
@@ -62,6 +62,61 @@ typedef double                          float64;
 
 #ifdef __cplusplus
 
+// used to limit the number of strings in memory
+// thread-safe and removes when no longer needed
+#include <boost/flyweight.hpp>
+
+class AssetHandle
```

Comment:
I think the term AssetHandle is getting too overloaded. Maybe call this an AssetKey?

Reply:
looks like AssetKey is present as well, maybe something like `AssetRefKey` ?

Reply:
Maybe just `AssetRef`?

The AssetKey definition in StorageManager can be removed and replaced with this class, so AssetKey is no longer needed there, but it looks like AssetKey is also used other places.